envoy_carrier.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+package lightstep.envoy;
+
+// CarrierStruct stores a set of key-values containing an OpenTracing span
+// context, as carried by Envoy, suitably encoded as a serialized protocol
+// buffer for transport via an HTTP header.
+//
+// The OpenTracing basictracer specifies the following keys and values:
+//
+//   "ot-tracer-traceid": 64 bit trace identifier encoded in 16 hex bytes
```

Comment:
just mentioning: I am enthusiastic about moving to a single key for the trace+span+sampled, and a single key for all baggage. but I don't want to take on that migration right now, so "just mentioning".